test/inference-engines/alarm.test.ts

```diff
@@ -1,206 +1,49 @@
-  expect(infer({ JOHN_CALLS: ['F'] }).toFixed(4)).toBe('0.9088')
-}
-
-const tests: { [key: string]: (engine: IInferenceEngine) => void } = {
-  'infers give Burglary True': infersAlarmGiveBurglaryTrue,
-  'infers give Burglary False': infersAlarmGiveBurglaryFalse,
-  'infers give Eath Quake True': infersAlarmGiveEathQuakeTrue,
-  'infers give Eath Quake False': infersAlarmGiveEathQuakeFalse,
-  'infers give Alarm True': infersAlarmGiveAlarmTrue,
-  'infers give Alarm False': infersAlarmGiveAlarmFalse,
-  'infers give John Calls True': infersAlarmGiveJohnCallsTrue,
-  'infers give John Calls False': infersAlarmGiveJohnCallsFalse,
-  'infers give Mary Calls True': infersAlarmGiveMaryCallsTrue,
-  'infers give Mary Calls False': infersAlarmGiveMaryCallsFalse,
-  'infers after set distribution': infersSetDistribution,
-}
-
-describe('infers', () => {
-  describe('alarm network', () => {
-    const testNames = Object.keys(tests)
-    const engine = new InferenceEngine(network)
-
-    for (const testName of testNames) {
-      const method = tests[testName]
-
-      it(`${testName}`, () => method(engine))
-    }
-  })
-})
+import { runTests } from './helpers'
+
+// These gold standard values were computed indepdendently in R.
+const GOLD_STANDARD: [string[], number][] = [
+  [['T', 'T', 'T', 'T', 'T'], 1.19700000E-06],
+  [['F', 'T', 'T', 'T', 'T'], 5.91015600E-04],
+  [['T', 'F', 'T', 'T', 'T'], 3.65034600E-04],
+  [['F', 'F', 'T', 'T', 'T'], 6.28111260E-04],
+  [['T', 'T', 'F', 'T', 'T'], 5.00000000E-11],
+  [['F', 'T', 'F', 'T', 'T'], 2.99400000E-08],
+  [['T', 'F', 'F', 'T', 'T'], 7.09290000E-07],
+  [['F', 'F', 'F', 'T', 'T'], 4.98002499E-04],
+  [['T', 'T', 'T', 'F', 'T'], 1.33000000E-07],
+  [['F', 'T', 'T', 'F', 'T'], 6.56684000E-05],
+  [['T', 'F', 'T', 'F', 'T'], 4.05594000E-05],
+  [['F', 'F', 'T', 'F', 'T'], 6.97901400E-05],
+  [['T', 'T', 'F', 'F', 'T'], 9.50000000E-10],
+  [['F', 'T', 'F', 'F', 'T'], 5.68860000E-07],
+  [['T', 'F', 'F', 'F', 'T'], 1.34765100E-05],
+  [['F', 'F', 'F', 'F', 'T'], 9.46204748E-03],
+  [['T', 'T', 'T', 'T', 'F'], 5.13000000E-07],
+  [['F', 'T', 'T', 'T', 'F'], 2.53292400E-04],
+  [['T', 'F', 'T', 'T', 'F'], 1.56443400E-04],
+  [['F', 'F', 'T', 'T', 'F'], 2.69190540E-04],
+  [['T', 'T', 'F', 'T', 'F'], 4.95000000E-09],
+  [['F', 'T', 'F', 'T', 'F'], 2.96406000E-06],
+  [['T', 'F', 'F', 'T', 'F'], 7.02197100E-05],
+  [['F', 'F', 'F', 'T', 'F'], 4.93022474E-02],
+  [['T', 'T', 'T', 'F', 'F'], 5.70000000E-08],
+  [['F', 'T', 'T', 'F', 'F'], 2.81436000E-05],
+  [['T', 'F', 'T', 'F', 'F'], 1.73826000E-05],
+  [['F', 'F', 'T', 'F', 'F'], 2.99100600E-05],
+  [['T', 'T', 'F', 'F', 'F'], 9.40500000E-08],
+  [['F', 'T', 'F', 'F', 'F'], 5.63171400E-05],
+  [['T', 'F', 'F', 'F', 'F'], 1.33417449E-03],
+  [['F', 'F', 'F', 'F', 'F'], 9.367427006190E-01],
+]
+
+const names: string[] = ['EARTHQUAKE', 'BURGLARY', 'ALARM', 'JOHN_CALLS', 'MARY_CALLS']
+const testValues: string[][][] = [
+  [['T'], ['F'], ['T', 'F']],
```

Comment:
Are all of these lines supposed to be the same?

Reply:
Yup.   This will test all possible combinations of the listed assignments to the variables.